tornado-api/src/main/java/uk/ac/manchester/tornado/api/TornadoExecutor.java

```diff
@@ -0,0 +1,202 @@
+package uk.ac.manchester.tornado.api;
+
+import uk.ac.manchester.tornado.api.common.TornadoDevice;
+import uk.ac.manchester.tornado.api.enums.TornadoVMBackendType;
+import uk.ac.manchester.tornado.api.exceptions.TornadoRuntimeException;
+import uk.ac.manchester.tornado.api.runtime.ExecutorFrame;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
```

Comment:
Can you please add the relevant Java doc that describes the object usage? I think we have it also in the `TornadoExecutionPlan`.